sample_app/app/views/layouts/application.html.erb

```diff
@@ -1,14 +1,18 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title><%= full_title(yield(:title))</title>
+    <title><%= full_title(yield(:title)) %></title>
     <%= csrf_meta_tags %>
-
-    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_link_tag    'application', media: 'all',
+                                              'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <%= render 'layouts/shim' %>
   </head>
-
   <body>
-    <%= yield %>
+    <%= render 'layouts/header' %>
+    <div class="container">
+      <%= yield %>
+	  <%= render 'layouts/footer' %>
```

Comment:
1)  The extra indenting on this `render` line for the footer is confusing here.  It should be at the same depth as `yield` since they are both children inside the `container` div.

2)  Do you want the footer inside the `container` div?  I'm guessing it should be outside it, mirroring the header.